From fdf172f68f2270306effda39211a4be5ca7e437e Mon Sep 17 00:00:00 2001
From: wangcichen <wangcichen_yewu@cmss.chinamobile.com>
Date: Tue, 17 May 2022 16:40:26 +0800
Subject: [PATCH 1/5] kpatch_process: fix possible double free.

---
 src/kpatch_process.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/kpatch_process.c b/src/kpatch_process.c
index 9d6daa4..a31f70c 100644
--- a/src/kpatch_process.c
+++ b/src/kpatch_process.c
@@ -682,7 +682,6 @@ process_list_threads(kpatch_process_t *proc,
 			t = realloc(pids, *alloc * sizeof(*pids));
 			if (t == NULL) {
 				kplogerror("Failed to (re)allocate memory for pids\n");
-				closedir(dir);
 				goto dealloc;
 			}
 
-- 
2.27.0

